### ISSUE TITLE
Wrong sanitization between inputBase and module ID

### ISSUE THREAD

**Issue by jangxyz** [rollup/rollup#5446]
### Rollup Version

v4.13.1

### Operating System (or Browser)

macos

### Node Version (if applicable)

v21

### Link To Reproduction

https://stackblitz.com/edit/rollup-repro-s7capj?file=projects%2F%5Bsubject%5D%2Frollup-sanitize%2Frollup.config.js&view=editor

### Expected Behaviour

While trying to bundle with an external dependency, I found that the build fails when the absolute path contains some sanitizable characters -- I had square brackets (`[]`) somewhere in the parent directory.

Setup:
- project lies in `~/project/[subject]/rollup-sanitize-1`
- build two bundles, app and lib, where app declares lib as external.

`
[!] RollupError: Invalid substitution "../../../../_subject_/rollup-sanitize-1/src/app/my_app" for placeholder "[name]" in "output.entryFileNames" pattern, can be neither absolute nor relative path.
`

(See screenshot below)

### Actual Behaviour

<details>

<summary>Screenshot</summary>

<img width="1567" alt="image" src="https://github.com/rollup/rollup/assets/52015/4bd09b7c-3722-46e0-9743-107bdbd6738f">

Screenshot of https://stackblitz.com/edit/rollup-repro-s7capj?file=projects%2F%5Bsubject%5D%2Frollup-sanitize%2Frollup.config.js&view=editor.

```bash
cd "projects/[subject]/rollup-sanitize-1"
npm install
npx rollup -c 
```

Note there is a separate project folder in `projects/_subject_/rollup-sanitize-2`, which works.

---

</details>

Digging through the code, I've noticed there is a mismatch between sanitized module ID and inputBase(*not sanitized*). The code is in [Chunk.ts](https://github.com/rollup/rollup/blob/fffaedeaa1cf9c8f6efc93d53bb8a81738e0ce87/src/Chunk.ts#L1071):

```typescript
class Chunk {
  getPreserveModulesChunkNameFromModule(module) {
    // ...
    const sanitizedId = sanitizeFileName(normalize(module.id.split(QUERY_HASH_REGEX, 1)[0]))
    // ...
    relative(this.inputBase, idWithoutExtension)
    // ...
  }
}
```

`inputBase` is assigned as a normal filepath (from `generateChunks` function in [Bundle.ts](https://github.com/rollup/rollup/blob/fffaedeaa1cf9c8f6efc93d53bb8a81738e0ce87/src/Bundle.ts#L176)) which is not sanizited, while `idWithoutExtension` has.

Hence while trying to find out the *relative path* from `"[subject]/rollup-sanitize-1/src/app/"` (not sanitized) to `"_subject_/rollup-sanitize-1/src/app/my_app"` (sanitized), it comes up with `"../../../../_subject_/rollup-sanitize-1/src/app/my_app"`

Technically, I could fix this on my side by:
* not making any parent directories with *unwise* characters, or
* passing `--no-sanitizeFileName` option.

Both approach are shown in the screenshot above, 
where on the right panel you are building on a separate project directory (`/_subject_/rollup-sanitize-2` instead of `[subject]/rollup-sanitize-1`),
and on the bottom you succeed with giving the `--no-sanitizeFileName` option.

I didn't know rollup was performing sanitization for the full path -- nor that it uses RFC 2396, and they declare some characters as 'unwise'([section 2.4.3](https://datatracker.ietf.org/doc/html/rfc2396#section-2.4.3)). It would probably make more sense that rollup applies `sanitizeFileName()` for both module ID *and* inputBase, before performing `relative()`?

**Comment by MIDZTECHNOLOGY** [comment]
Please I would like to work on this

**Comment by lukastaegert** [comment]
Yes, I agree this should be symmetric between inputBase and the ids, so inputBase should be sanitized too. PR welcome, @MIDZTECHNOLOGY!
